geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Transform.scala

```diff
@@ -0,0 +1,248 @@
+   *
+   * @param name name
+   * @param binding binding
+   * @param original name of the original attribute being projected
+   * @param i index of the original attribute being projected
+   */
+  case class RenameTransform(name: String, binding: Class[_], original: String, i: Int) extends Transform {
+    override def evaluate(feature: SimpleFeature): AnyRef = feature.getAttribute(i)
+  }
+
+  /**
+   * An attribute comprised of an expression (function, math transform, json-path, etc)
+   *
+   * @param name name
+   * @param binding binding
+   * @param expression expression
+   */
+  case class ExpressionTransform(name: String, binding: Class[_], expression: Expression) extends Transform {
+    override def evaluate(feature: SimpleFeature): AnyRef = expression.evaluate(feature)
+  }
+
+  object Transforms {
+
+    /**
+     * Create transform definitions for a delimited expression string
+     *
+     * @param sft original simple feature type
+     * @param transforms semi-colon delimited transform expressions
+     * @return
+     */
+    def apply(sft: SimpleFeatureType, transforms: String): Seq[Transform] =
+      if (transforms.isEmpty) { Seq.empty } else { apply(sft, transforms.split(DefinitionDelimiter)) }
+
+    /**
+     * Create transform definitions for a sequence of expression strings
+     *
+     * @param sft original simple feature type
+     * @param transforms transform expressions
+     * @return
+     */
+    def apply(sft: SimpleFeatureType, transforms: Seq[String]): Seq[Transform] =
+      transforms.map(definition(sft, _))
+
+    /**
+     * Create a transform definition from an expression
+     *
+     * @param sft simple feature type
+     * @param transform transform expression
+     * @return
+     */
+    private def definition(sft: SimpleFeatureType, transform: String): Transform = {
+      val equals = transform.indexOf('=')
+      if (equals == -1) {
+        val name = transform.trim()
+        val i = sft.indexOf(name)
+        if (i == -1) {
+          attributeExpression(sft, name)
+        } else {
+          PropertyTransform(name, sft.getDescriptor(i).getType.getBinding, i)
+        }
+      } else {
+        val name = transform.substring(0, equals).trim()
+        val exp = transform.substring(equals + 1).trim()
+        val expression = try { ECQL.toExpression(exp) } catch {
+          case NonFatal(e) => throw new IllegalArgumentException(s"Unable to parse expression $transform:", e)
+        }
+        expression match {
+          case f: Function =>
+            ExpressionTransform(name, f.getFunctionName.getReturn.getType, f)
+
+          case f: MathExpressionImpl =>
+            ExpressionTransform(name, classOf[java.lang.Double], f) // math ops always return doubles?
+
+          case p: PropertyName =>
+            val orig = p.getPropertyName
+            val i = sft.indexOf(orig)
+            if (i == -1) {
+              attributeExpression(sft, orig).copy(name = name)
```

Comment:
Same here, What's an example of a string that hits this branch?

Reply:
same thing as before, but in this case the attribute is renamed. previous statement would be `"$.foo.bar"`, this one would be `"baz=$.foo.bar"`